### ISSUE TITLE
extend the interface of GlobalAlgBlk to return the full decision word(s)

### ISSUE THREAD

**Issue by fwyzard** [cms-l1t-offline/cmssw#289]
As far as I know, the way to access the L1 results is via something like

``` c++
  // get the L1 results
  edm::Handle<GlobalAlgBlkBxCollection> ugt;
  event.getByToken(ugt_token_, ugt);
  GlobalAlgBlk const & results = ugt->at(0);

  // get the individual decisions from the GlobalAlgBlk
  for (unsigned int i = 0; i < GlobalAlgBlk::maxPhysicsTriggers; ++i)
    std::cerr << "bit: " << i << "\tresult: " << result.getAlgoDecisionFinal(i) << std::endl;
```

the last part is unnecessarily expensive, because `getAlgoDecisionFinal(i)`

``` c++
bool GlobalAlgBlk::getAlgoDecisionFinal(unsigned int bit)  const
{  
   if(bit>=m_algoDecisionFinal.size()) return false;
   return m_algoDecisionFinal.at(bit);
}
```

 will check that `bit` is valid twice per call: once explicitly, the second time implicitly within the `at(...)` call.

Would it be possible to add a method to access the whole decision directly, via something like

``` c++
std::vector<bool> const & GlobalAlgBlk::getAlgoDecisionFinalWord() const {
    return m_algoDecisionFinal;
}
```

?

**Comment by mulhearn** [comment]
Seems harmless...

**Comment by mulhearn** [comment]
If you add this to your PR that uses it, I'll sign off.  Or we can add it ourselves first.  Whatever you prefer.

**Comment by blwiner** [comment]
This seems like a straight forward addition. (I'm a little surprised we
don't already have it...but we don't).  How do you want to proceed?

On Mon, May 2, 2016 at 8:31 PM, mulhearn notifications@github.com wrote:

> If you add this to your PR that uses it, I'll sign off. Or we can add it
> ourselves first. Whatever you prefer.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/cms-l1t-offline/cmssw/issues/289#issuecomment-216404060

## 

# 

| Prof. Brian L. Winer |
| --- |

```
              Department of Physics
           The Ohio State University
             191 W. Woodruff Ave
              Columbus, OH 43210

               (614) 292-8996 (Office)
               (614) 292-7370 (Lab)
```

#

**Comment by fwyzard** [comment]
fixed by cb703f8 (for 80x) and a34f1c5 (for 81x)
